### DISCUSSION TITLE
Mindspace (open source mind mapping tool)

### DISCUSSION THREAD

**Discussion by canadaduane** [bikeshaving/crank#263] [Show and tell]
This is my first project using Crank.js, and I'm very excited by and satisfied with the ease and flexibility that the framework offers. 

Honestly, it was so refreshing when compared with React's hooks and other oddities that are now mainstream. Crank's simple generator-based system made it so easy to reason about. Also, the event system was sensible and helpful when sub-components needed to communicate with parents. Lastly, the ability to use template string literals was delightful, as it meant I did not need a compiler--the app was built with plain HTML and (ESM) javascript.

About the app:

"For me, mind mapping is as much an emotional exercise as it is reasoning about a representation of knowledge. Ideas and concepts are laid out, grouped, discarded, connected, disconnected.

"I wanted a UI that encourages treating thoughts and concepts as toys to play with. If you're looking for something similar, you will probably enjoy Mindspace even in its early phase."

**Repo**: https://github.com/canadaduane/mindspace

**Demo**: https://halecraft.org/mindspace/

**Comment by darrylcousins** [comment]
Hi Duane and Kelty,

Mindspace is indeed a beautiful interface.

Your comments about Crank align with my own thoughts.

Ngā mihi,
Darryl
